icu4c/source/i18n/nfrule.cpp

```diff
@@ -153,6 +153,7 @@ NFRule::makeRules(UnicodeString& description,
         // base value is an even multiple of its divisor (or it's one
         // of the special rules)
         if ((rule1->baseValue > 0
+            && (rule1->radix != 0) // ICU-23109 Ensure next line won't "% 0"
```

Comment:
Is a radix of zero even legal, or should it have produced an error earlier, when parsing the rule? (I don't know the answer to this question).

Related, at  line 120, there is no checking of the error code when creating the rule.